### PR TITLE
feat: signal which products are quota limited

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

In https://github.com/PostHog/posthog/pull/16725 I changed capture so that we could have posthog-js clients not continue sending session recordings when they were quota limited

But, that meant all older clients would retry traffic that they previously wouldn't have done.

We had to revert that, but still need to reduce load from recordings clients

## Changes

* Still return a 200 when dropping events due to quota limiting
* If recordings were quota limited then return a list of limited products in that request as part of the payload

So, the possible responses _now_ are

`{"status": 1}`

and

`{"status": 1, "quotaLimited": ["recordings"]}`

I considered more structure in the payload

e.g.

```
{
    "status": 1,
    "quota_limited": {
        "recordings": {
             "retry-after": 60
        }
    }
}
```

That would mean we have space to amend this in the future, but this is the first change to response shape in 2 or 3 years, so, it's reasonable to assume that it won't change much from here.

## How did you test this code?

developer tests

### CODE REVIEW

File: posthog/api/capture.py [hunk 1]
```diff
@@ -481,7 +500,13 @@ def get_event(request):
         pass
 
     statsd.incr("posthog_cloud_raw_endpoint_success", tags={"endpoint": "capture"})
-    return cors_response(request, JsonResponse({"status": 1}))
+
+    response_payload: Dict[str, Any] = {"status": 1}
+
+    if recordings_were_quota_limited:
+        response_payload["quotaLimited"] = ["sessionRecordings"]
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/api/capture.py:507]: Ah, the python JSON format strikes again 🤣 

I had it as `sessionRecordings` to match the batch key in the JS but there's nothing stopping us changing that so they match all the way through 👍

**Comment by pauldambra** [comment]
pairs with https://github.com/PostHog/posthog-js/pull/765

**Review by benjackwhite — APPROVED** [review]
200 lgtm

File: ee/api/test/test_capture.py [hunk 2]
```diff
@@ -162,3 +211,71 @@ def test_partition_key_override(self, kafka_produce):
...
+            from ee.billing.quota_limiting import QuotaResource, replace_limited_team_tokens
+
+            replace_limited_team_tokens(
+                QuotaResource.RECORDINGS, {self.team.api_token: timezone.now().timestamp() + 10000}
+            )
+            _, response = self._send_session_recording_event()
+
+            assert response.json() == {
+                "status": 1,
+                "quotaLimited": ["recordings"],
```
Comment (pauldambra) [inline ee/api/test/test_capture.py:269]:
```suggestion
                "quota_limited": ["recordings"],
```

**Pull request closed without merging**